src/main/java/Barbie.java

```diff
@@ -0,0 +1,182 @@
+import types.Deadlines;
+import types.Party;
+import types.Task;
+import types.Todo;
+import exceptions.*;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+
+public class Barbie {
+
+    enum Command {
+        MARK,
+        UNMARK,
+        DEL,
+        TODO,
+        DEADLINE,
+        PARTY,
+        LIST,
+        BYE,
+    }
+
+
+    public static void main(String[] args) {
+        // CONSTANTS
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Task> list = Storage.getLastList();
+        Path path = Paths.get("barbie.txt");
+        int indexNumber = list.size(); // Starting from 1 reduces the need to subtract and add 1 for usability.
+
+
+        // Intro
+        Ui.intro(Utils.getDateList(LocalDate.now(), list));
+
+            loop:
+            while (true) {
+                try {
+
+                    String input = scanner.nextLine();
+                    String[] parts = input.split(" ", 2);
+                    Command command = Command.valueOf(parts[0].toUpperCase());
+
+                    Ui.barbie();
+
+                    switch (command) {
```

Comment:
The main class, Barbie, seems to be handling many responsibilities. Consider using a command class to break down the responsibilities.